deployments/azure-kubernetes-service.yml

```diff
@@ -42,39 +57,47 @@ jobs:
       uses: azure/login@v1
       with:
         creds: ${{ secrets.AZURE_CREDENTIALS }}
-    
-    - name: Build image on ACR
-      uses: azure/CLI@v1
+
+    - name: Login to ACR
+      uses: Azure/docker-login@v1
       with:
-        azcliversion: 2.29.1
-        inlineScript: |
-          az configure --defaults acr=${{ env.AZURE_CONTAINER_REGISTRY }}
-          az acr build -t  -t ${{ env.AZURE_CONTAINER_REGISTRY }}.azurecr.io/${{ env.PROJECT_NAME }}:${{ github.sha }}
+        login-server: ${{ env.AZURE_CONTAINER_REGISTRY }}.azurecr.io
+        username: ${{ secrets.REGISTRY_USERNAME }}
+        password: ${{ secrets.REGISTRY_PASSWORD }}
+      
+    - name: Build and push docker image to ACR
+      run: |
+        docker build . -t ${{ env.AZURE_CONTAINER_REGISTRY }}.azurecr.io/${{ env.CONTAINER_NAME }}:${{ github.sha }}
+        docker push ${{ env.AZURE_CONTAINER_REGISTRY }}.azurecr.io/${{ env.CONTAINER_NAME }}:${{ github.sha }}
     
     - name: Gets K8s context
-      uses: azure/aks-set-context@v1
+      uses: azure/aks-set-context@v2.0
       with:
-          creds: ${{ secrets.AZURE_CREDENTIALS }}
           resource-group: ${{ env.RESOURCE_GROUP }}
           cluster-name: ${{ env.CLUSTER_NAME }}
-      id: login
+
+    - name: Create K8s secret for pulling image from ACR
+      uses: Azure/k8s-create-secret@v1.1
+      with:
```

Comment:
Creating secret in AKS for pulling images from ACR was intentionally avoided earlier. There was effort done where both AKS and ACR collaborated to have trust between AKS and ACR by default.

If this is the recommendation from ACR and AKS team then I am good with the change.